### PR TITLE
refactor(frontend): hard-code Coingecko ENV variables

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

The URL for Coingecko API can be removed from the .env files and hard-coded, since it is constant for all environment.

That said, we move all Coingecko-related "env" variables to a specific module in `/env` folder

# Changes

- Hard-code Coingecko API URL.
- Move API URL and KEY variables to an `env` module.
- Adapt the code.
- Remove the API URL from the template `.env` files.

### CODE REVIEW

File: src/frontend/src/env/rest/coingecko.env.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { UrlSchema } from '$lib/validation/url.validation';
+
+export const COINGECKO_API_URL = UrlSchema.parse('https://pro-api.coingecko.com/api/v3/');
```
Comment (peterpeterparker) [inline src/frontend/src/env/rest/coingecko.env.ts:3]: `parse` can throw an error at runtime which can per extension lead to the all app to not mount since this is executed at the root of the module.

We should probably rather use `safeParse`?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: my idea is that since they are hard-coded, the error would appear safely with CIs failing, in any PR that changes/updates/modifies it

so we would be aware of it, before merging a possible not-valid-URL

with `safeParse`, if the URL is wrongly parsed, the app would work either way, but only the `COINGECKO_API` functions will possibly fail

Comment (peterpeterparker) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: How the CI would fail? `safe` is a runtime feature.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: What I meant was that with `parse` will fail, and that would be good

### `parse`

- CI runs and find an error in parsing --> fails --> we know that the URL we wrote in the code is somehow wrong

### `safeParse`

```typescript
export const COINGECKO_API_URL = UrlSchema.safeParse(...).success ? ... : fallbackValue;
```

- CI runs --> there is an error in parsing, but it falls back on the fallback value --> we think it is all good, but the Coingecko features are not working


Or you mean to not have a fallback value but raise the error? Maybe I misunderstood what you meant as implementation

NOTE: with E2E tests obviously this is not a discussion eheh

Comment (peterpeterparker) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: I guess the CI don't run E2E tests against the production build, hence the CI won't run the production application and it can potentially ends with an application that does not start / mount.

It would be an edge case but, it's still a possible case therefore I think it's not safe to actually use `parse` at the root of a module / of the application without catching potential errors.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: yes, makes sense! What do you think would be better then? Throw an error in case of parsing unsuccessful or just fallback to an empty string?

Comment (peterpeterparker) [reply src/frontend/src/env/rest/coingecko.env.ts:3]: What's the current behavior if someone forgot to setup an env variable, empty string?
Should we just keep that approach (for the best or worst) and iterate if necessary?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts]: it would default to an `undefined`, so I keep the same; have a look at last commit

Comment (peterpeterparker) [reply src/frontend/src/env/rest/coingecko.env.ts]: > it would default to an undefined, so I keep the same

I'm confuse, current code default to `''`

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts]: very sorry, I forgot to push:

- Before: if no env variable is set, it is `undefined`;
- After: if `safeParse` fails, it falls back on `undefined`

File: src/frontend/src/env/rest/coingecko.env.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { UrlSchema } from '$lib/validation/url.validation';
+
+const parsedCoingeckoApiUrl = UrlSchema.safeParse('https://pro-api.coingecko.com/api/v3/');
```
Comment (peterpeterparker) [inline src/frontend/src/env/rest/coingecko.env.ts:3]: If you want fancyness `const {success, data} = ...` 😉

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: src/frontend/src/env/rest/coingecko.env.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/env/rest/coingecko.env.ts]: Thx

File: src/frontend/src/env/rest/coingecko.env.ts (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/env/rest/coingecko.env.ts]: good one!
